src/infrastructure/database/models/user/user_database_handler.py

```diff
@@ -208,6 +235,75 @@ async def remove_from_favourite_counter(
             }
         )
 
+    @staticmethod
+    async def send_password_reset_request(
+            payload: UserEmail,
+            collection: Collection[User],
+            user: User
+    ):
+        try:
+            settings = get_settings()
+            token = randbytes(10)
+            change_password_code = hash_token(token)
+            collection.find_one_and_update({'_id': user['_id']}, {
+                '$set': {'reset_password_code': change_password_code, 'updated_at': datetime.utcnow()}},
+                                           return_document=ReturnDocument.AFTER)
+            url = f'https://{settings.WEB_HOST}/reset_password/{token.hex()}'
+            await Email(user, url, [EmailStr(payload.email)]).send_reset_password_code()
+        except Exception:
+            collection.find_one_and_update({'_id': user['_id']},
+                                           {'$set': {'reset_password_code': None, 'updated_at': datetime.utcnow()}},
+                                           return_document=ReturnDocument.AFTER)
+            raise SendingEmailError()
+
+    @staticmethod
+    async def check_reset_token(
+            token: str,
+            collection: Collection[User]
+    ):
+        reset_password_code = dehash_token(token)
+        return collection.find_one({'reset_password_code': reset_password_code})
+
+    @staticmethod
+    async def change_password(
+            new_password: str,
+            token: str,
+            collection: Collection[User]
+    ):
+        change_password_code = dehash_token(token)
+        password_salt = gensalt().decode('utf-8')
+        new_password = UserDatabaseHandler.get_password_hash(new_password, password_salt)
+        collection.find_one_and_update({'reset_password_code': change_password_code},
+                                       {'$set': {'password': new_password, 'password_salt': password_salt,
+                                                 'change_password_code': None, 'updated_at': datetime.utcnow()}})
+
+    @staticmethod
+    async def send_deletion_request(
+            current_user: User,
+            collection: Collection[User]
+    ):
+        settings = get_settings()
+        try:
+            token = randbytes(10)
+            delete_account_code = hash_token(token)
```

Comment:
tak na dobrą sprawę to nie powinno być tak mieszane..
Z założenia UserDatabaseHandler ma załatwiać tylko sprawy związane z bazą, wysyłanie maili etc powinno być gdzie indziej

Reply:
Tak tylko informacyjnie